core/services/chainlink/config_dump.go

```diff
@@ -527,6 +529,21 @@ func (c *Config) loadLegacyCoreEnv() {
 		Headers:        (*[]audit.ServiceHeader)(audit.AuditLoggerHeaders.ParsePtr()),
 	}
 
+	var lockEnabled *bool
+	if mode := envvar.NewString("DatabaseLockingMode").ParsePtr(); mode == nil { // dual default
+		lockEnabled = nil // lease default
+	} else {
+		switch *mode {
+		case "advisorylock":
+			return fmt.Errorf("%w: '%s' mode: must use 'lease' or 'none'", config.ErrUnsupported, *mode)
+		default:
+			return fmt.Errorf("%w: unrecognized mode '%s': must use 'lease' or 'none'", config.ErrUnsupported, *mode)
+		case "none":
+			lockEnabled = ptr(false)
+		case "lease", "dual":
+			lockEnabled = nil // lease default
```

Comment:
```suggestion
		case "none":
			lockEnabled = ptr(false)
		case "lease", "dual":
			lockEnabled = nil // lease default
		default:
			return fmt.Errorf("%w: unrecognized mode '%s': must use 'lease' or 'none'", config.ErrUnsupported, *mode)
```

I dunno if there is a style guide on this but to my eye it seems cleaner to have the default at the bottom.